### PR TITLE
Fail tests on E_DEPRECATED

### PR DESCRIPTION
**Pull request by mpesari**
## Related tickets & documents

PHP 8.2 is stable and should be supported.

Some tests emitted `E_DEPRECATED` in PHP 8.2 due to use of dynamic properties, but it was fixed.

This PR makes phpunit to fail tests which emit `E_DEPRECATED` in order to be able to fix deprecated features before they are removed from a new version of PHP.

## Description

Deprecation notices (in phpunit) were fixed by https://github.com/paytrail/paytrail-php-sdk/pull/59

### CODE REVIEW

**Comment by mpesari** [comment]
I'm not sure how the failing docker push is supposed to work here... but it works in my fork: https://github.com/mpesari/paytrail-php-sdk/actions/runs/4009270119

**Comment by VuorelaJ-P** [comment]
Cold be this
https://stackoverflow.com/questions/70646920/github-token-permission-denied-write-package-when-build-and-push-docker-in-githu
Other site suggested that you have
```
permissions:
      contents: read
      packages: write
```
Don't know why it fails only in PHP 8.2, but I had same problem when tried to add it as experimental. Which why it hasn't been there yet.

**Comment by mpesari** [comment]
OK so the `php-actions/phpunit` action is trying to build and push a Docker image:

```
Pulling docker.pkg.github.com/paytrail/paytrail-php-sdk/php-actions_composer_paytrail-php-sdk:php-8.2-build2.1.0
  Error response from daemon: manifest unknown
  Dockerfile:
  FROM php:8.2-cli-alpine
...
  Successfully built 2dd6c0eac65a
  Successfully tagged docker.pkg.github.com/paytrail/paytrail-php-sdk/php-actions_phpunit_paytrail-php-sdk:php-8.2-build2.1.0
  The push refers to repository [docker.pkg.github.com/paytrail/paytrail-php-sdk/php-actions_phpunit_paytrail-php-sdk]
  denied: installation not allowed to Write organization package
```

In other PHP version cases the image is already built and present. Dunno how.

I did enable this setting in my fork repository:

 * Settings > Actions > General > Workflow permissions > Read and write permissions

**Comment by mpesari** [comment]
There is an option to require approval before running workflows from outside forks, maybe it will be enough?

**Comment by VuorelaJ-P** [comment]
Yeah, must be some sort of fork problem. Since permissions are correct on repository settings, but somehow forks does not work properly.
Problems is that this would probably lead all future forks failing to run on 8.2, which is huge problem.

**Comment by VuorelaJ-P** [comment]
Managed to make 8.2 working in #59 
There's some deprecated methods, which can be removed later. This PR might get closed, after that one is merged. But we can use this first to make sure that pipeline success on this as well. If not, we need to fix something else as well.

**Comment by VuorelaJ-P** [comment]
@mpesari Can you please rebase this branch to master, and see if checks are running properly? At least now new branches and master will run 8.2 properly, but we want to make sure requests from other repositories will run checks as well without any issues.

**Comment by mpesari** [comment]
Nice! Pipeline seems to work now. I updated the title and description to be more.. descriptive

**Comment by VuorelaJ-P** [comment]
It might be that newer versions are required to being exist on master. Or its just those extra permissions definitions to workflow file.

**Review by VuorelaJ-P — APPROVED** [review]

**Pull request merged**
